api/server/sdk/volume_ops.go

```diff
@@ -352,6 +352,50 @@ func (s *VolumeServer) Stats(
 	}, nil
 }
 
+func (s *VolumeServer) CapacityUsage(
+	ctx context.Context,
+	req *api.SdkVolumeCapacityUsageRequest,
+) (*api.SdkVolumeCapacityUsageResponse, error) {
+
+	if len(req.GetVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply volume id")
+	}
+
+	dResp, err := s.driver().CapacityUsage(req.GetVolumeId())
+	if err != nil {
+		return nil, status.Errorf(
+			codes.Internal,
+			"Failed to obtain stats for volume %s: %v",
+			req.GetVolumeId(),
+			err.Error())
+	}
+
+	resp := &api.SdkVolumeCapacityUsageResponse{}
+	resp.CapacityUsageInfo = &api.CapacityUsageInfo{}
+	if dResp != nil {
+		resp.CapacityUsageInfo.ExclusiveBytes = dResp.CapacityUsageInfo.ExclusiveBytes
+		resp.CapacityUsageInfo.SharedBytes = dResp.CapacityUsageInfo.SharedBytes
+		resp.CapacityUsageInfo.TotalBytes = dResp.CapacityUsageInfo.TotalBytes
+		if dResp.ErrorDetails != "" {
```

Comment:
Instead of comparing strings, let's make this an `error` val.

Reply:
done.